### PR TITLE
Add basic support for theming!

### LINKED ISSUE #19857: Add theming capabilities
We want to allow the end users of Directus to pick and choose what themes they want to use in app, create their own themes, and allow them to override individual rules of the theme. This unlocks way more white-labeling capabilities and should reduce the need to rely on CSS overrides.
```[tasklist]
### Tasks
- [ ] https://github.com/directus/directus/issues/19916
- [ ] https://github.com/directus/directus/issues/19969
```

### PR DESCRIPTION
**Pull request by rijkvanzanten**
## Scope

What's changed:

- Renamed "Theme" to "Appearance"
- Added project-level default appearance setting
- Added a new `@directus/themes` package
  - Added a use-theme composable that combines the chosen theme with defaults and overrides
  - Added a theme-provider component that injects the css variables and fonts in the head
  - Added a use-fonts composable that'll generate google fonts to load
- Added the theme-provider to the Directus app
  - Added support for loading custom google fonts
- Updated a lot of the used variables to use the new theme based variables
- Added a new system interface to configure custom theme rules for customizability
- Added support to use the Project Color as the primary color system wide
- Added support for custom favicons

https://github.com/directus/directus/assets/9141017/0c2b4c5f-c7a6-4eff-b68f-78438f71bf3c

## Potential Risks / Drawbacks

- Not all exact rules will be available yet
- Some colors might be a tad wonky due to using new variables

## Review Notes / Questions

- Notably this does not include multiple system themes to pick from. The infrastructure exists for it though, so we can enable it easily once more rules and default themes are available.
- The next iteration will include themes as an extension type, and multiple-shipped-ootb themes

---

Fixes #19857

### CODE REVIEW

**Comment by paescuj** [comment]
Awesome work 🥇 (Not even that bad with Comic Sans 🤣)

File: .changeset/tough-schools-swim.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+---
+'@directus/app': minor
+---
+
+Add support for uploading custom favicon files
```
Comment (azrikahar) [inline .changeset/tough-schools-swim.md:5]:
```suggestion
Added support for uploading custom favicon files
```

File: api/src/database/migrations/20230927A-themes.ts [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+	/**
+	 * Knex doesn't support setting defaults to null (you'll end up with `NULL::character varying`),
+	 * so we'll have to create a new column, copy over the relevant bits, and remove the old
+	 */
+	await knex.schema.alterTable('directus_users', (table) => {
+		table.string('appearance');
+	});
+
+	await knex('directus_users').update({ appearance: 'dark' }).where({ theme: 'dark ' });
+	await knex('directus_users').update({ appearance: 'light' }).where({ theme: 'light ' });
```
Comment (azrikahar) [inline api/src/database/migrations/20230927A-themes.ts:13]:
```suggestion
	await knex('directus_users').update({ appearance: 'dark' }).where({ theme: 'dark' });
	await knex('directus_users').update({ appearance: 'light' }).where({ theme: 'light' });
```
assuming these spaces aren't intended, at least from what I can observe with our existing theme behavior

File: packages/types/src/settings.ts
Comment (azrikahar) [inline packages/types/src/settings.ts:1]: do we need to add `public_favicon` in `Settings` type as well?

**Review by azrikahar — COMMENTED** [review]
tiny suggestions/questions from what I can observe 😄

**Pull request merged**